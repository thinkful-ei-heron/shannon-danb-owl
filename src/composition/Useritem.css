.participantInfo {
  list-style:none;
  padding: 0;
  text-align: left;
  position: relative;
  left: 55px;
  top: 13px;
}

.avatar {
  width: 50px;
  position: absolute;
  left: 0;
  border-radius: 50%;
}

.greenCircle {
  margin-right: 5px;
  height: 15px;
  width: 15px;
  display: inline-block;
  background-color: green;
  border-radius: 50%;
}

.greyCircle {
  margin-right: 5px;
  height: 15px;
  width: 15px;
  display: inline-block;
  background-color: gray;
  border-radius: 50%;
}
