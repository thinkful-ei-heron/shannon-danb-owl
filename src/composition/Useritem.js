import React from 'react';
import './Useritem.css';

function Useritem(props){
  return (
    <div className= "eachParticipantInfo">
      <img src= {props.avatar} alt= 'avatar' className= 'avatar' />
      <ul className= "participantInfo">
        <li>{props.name}</li>
        <li><span className= {props.onStage || props.inSession ? 'greenCircle' : 'greyCircle'}></span>{props.onStage ? 'on stage' : props.inSession ? 'in session' : 'left session'}</li>
      </ul>
  </div>
  );
}


export default Useritem;