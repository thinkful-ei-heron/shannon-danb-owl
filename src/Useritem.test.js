import React from 'react';
import ReactDOM from 'react-dom';
import Useritem from './composition/Useritem';
import renderer from 'react-test-renderer';


describe ('rendering Useritem', () => {
  it ('renders without crashing', () => {
    const div= document.createElement('div');
    ReactDOM.render(<Useritem 
      key={1}
      avatar='https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'
      name='Johnny'
      onStage='true'
      inSession= 'false' 
      />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

it ('renders correctly (onStage is true, in Session is false)', () => {
  const tree= renderer
  .create(<Useritem
    key={1}
    avatar='https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'
    name='Johnny'
    onStage='true'
    inSession= 'false'  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

it ('renders correctly (onStage and inSession are both false)', () => {
  const tree= renderer
  .create(<Useritem
    key={1}
    avatar='https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'
    name='Johnny'
    onStage='false'
    inSession= 'false'  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

it ('renders correctly (onStage is false, inSession is true)', () => {
  const tree= renderer
  .create(<Useritem
    key={1}
    avatar='https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'
    name='Johnny'
    onStage='false'
    inSession= 'true'  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});


})