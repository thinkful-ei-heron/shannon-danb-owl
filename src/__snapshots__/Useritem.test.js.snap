// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Useritem renders correctly (onStage and inSession are both false) 1`] = `
<div
  className="eachParticipantInfo"
>
  <img
    alt="avatar"
    className="avatar"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <ul
    className="participantInfo"
  >
    <li>
      Johnny
    </li>
    <li>
      <span
        className="greenCircle"
      />
      on stage
    </li>
  </ul>
</div>
`;

exports[`rendering Useritem renders correctly (onStage is false, inSession is true) 1`] = `
<div
  className="eachParticipantInfo"
>
  <img
    alt="avatar"
    className="avatar"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <ul
    className="participantInfo"
  >
    <li>
      Johnny
    </li>
    <li>
      <span
        className="greenCircle"
      />
      on stage
    </li>
  </ul>
</div>
`;

exports[`rendering Useritem renders correctly (onStage is true, in Session is false) 1`] = `
<div
  className="eachParticipantInfo"
>
  <img
    alt="avatar"
    className="avatar"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <ul
    className="participantInfo"
  >
    <li>
      Johnny
    </li>
    <li>
      <span
        className="greenCircle"
      />
      on stage
    </li>
  </ul>
</div>
`;
